src/ILLink.RoslynAnalyzer/RequiresAnalyzerBase.cs

```diff
@@ -341,15 +341,16 @@ private void ReportMismatchInAttributesDiagnostic (SymbolAnalysisContext symbolA
 		/// <summary>
 		/// True if the source of a call is considered to be annotated with the Requires... attribute
 		/// </summary>
-		protected bool IsMemberInRequiresScope (ISymbol containingSymbol)
+		protected bool IsMemberInRequiresScope (ISymbol member)
 		{
-			if (containingSymbol.HasAttribute (RequiresAttributeName) ||
-				containingSymbol.ContainingType.HasAttribute (RequiresAttributeName)) {
+			if (member.HasAttribute (RequiresAttributeName) ||
+				member is not ITypeSymbol &&
```

Comment:
Nit - I would prefer parenthesis around the `( && )` I know that this works, but I think it would be more readable that way.

Reply:
I will add parenthesis to both Analyzer and linker